opensrp-reveal/src/main/java/org/smartregister/reveal/server/FileHTTPServer.java

```diff
@@ -0,0 +1,157 @@
+package org.smartregister.reveal.server;
+
+import org.apache.commons.lang3.StringUtils;
+import org.smartregister.reveal.BuildConfig;
+import org.smartregister.reveal.application.RevealApplication;
+import org.smartregister.util.Utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+/**
+ * A minimal HTTP server that serves a static file (e.g mapbox style json file)
+ */
+public class FileHTTPServer {
+    public static final int PORT = 9783;
+
+    private static final String DEFAULT_STYLE_JSON_FILE = "map-download-style.json";
+    private static final String DEFAULT_DG_ID_PLACEHOLDER = "DIGITAL_GLOBE_ID";
+    private static String styleJson;
+
+    private final FileHTTPServer.ServerThread server;
+    private final ServerSocket socket;
+
+    public FileHTTPServer(String styleJsonFile, String  digitalGlobeIdPlaceHolder) throws IOException {
+        socket = createBoundSocket(PORT);
+        if (socket == null) {
+            throw new IOException("Could not find an available port");
+        }
+        server = new ServerThread(socket);
+
+        digitalGlobeIdPlaceHolder = StringUtils.isNotBlank(digitalGlobeIdPlaceHolder) ? digitalGlobeIdPlaceHolder : DEFAULT_DG_ID_PLACEHOLDER;
+        styleJsonFile = StringUtils.isNotBlank(styleJsonFile) ? styleJsonFile : DEFAULT_STYLE_JSON_FILE;
+
+        styleJson = Utils.readAssetContents(RevealApplication.getInstance().getApplicationContext(), styleJsonFile);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4217760)